pkg/api/v1beta3/dynakube/feature_flags.go

```diff
@@ -247,6 +251,36 @@ func (dk *DynaKube) FeatureMaxFailedCsiMountAttempts() int {
 	return maxCsiMountAttemptsValue
 }
 
+func (dk *DynaKube) FeatureMaxCSIRetryTimeout() time.Duration {
+	maxCsiMountTimeoutValue := dk.getFeatureFlagRaw(AnnotationFeatureMaxCsiMountTimeout)
+
+	duration, err := time.ParseDuration(maxCsiMountTimeoutValue)
+	if err != nil || duration < 0 {
+		duration, _ = time.ParseDuration(DefaultMaxCsiMountTimeout)
+	}
+
+	return duration
+}
+
+// MountAttemptsToTimeout converts the (old) number of csi mount attempts into a time.Duration string.
+// The converted value is based on the exponential backoff's algorithm.
+// The output is string because it's main purpose is to convert the value of an annotation to another annotation.
+func MountAttemptsToTimeout(maxAttempts int) string {
+	var power float64 = 2
+
+	var baseDelay = 1 * time.Second / 2
+
+	var delaySum time.Duration
+
+	for i := range maxAttempts {
+		secRetry := math.Pow(power, float64(i))
+		delay := time.Duration(secRetry) * baseDelay
+		delaySum += delay
```

Comment:
are you sure you need loop and can't rewrite using formula?

Reply:
looking at
```
	baseDelay := time.Second / 2
	delay := time.Duration(math.Exp2(float64(maxAttempts))) * baseDelay
```

seems to work

Reply:
![image](https://github.com/user-attachments/assets/9e1e91b7-d587-4c18-add0-9a05d76a1ffd)

Reply:
(fixed)

Reply:
I am confused, do we now need a loop for this calculation or not? You committed a solution without a loop, but reading this conversation I would assume we need a loop

Reply:
you don't need loop, as I originally suggested you can rewrite with formula, and @0sewa0 did it.